### ISSUE TITLE
Test-Connection hangs when used in a Windows form

### ISSUE THREAD

**Issue by NJK-HOME** [PowerShell/PowerShell#12021]
# Steps to reproduce
Create a simple Windows form that utilizes the "Test-Connection" cmdlet.  In practice the form is more complex - but the example below provides a simple test to reproduce the issue and eliminate extra complexity.

```powershell
Add-Type -AssemblyName System.Windows.Forms
$Form = New-Object system.Windows.Forms.Form
$Form.Text = "Sample Form"
$Button = New-Object Windows.Forms.Button
$Button.Text = "Test"
$Button.Location = New-Object Drawing.Point(100,100)
$Button.Add_Click({
    $Button.Text = "Running..."
    $Button.Enabled = $false
    Test-Connection $env:COMPUTERNAME -Count 3 -Quiet
    $Button.Text = "Test"
    $Button.Enabled = $true
})
$Form.Controls.Add($button)
$Form.ShowDialog()
```

# Expected behavior
Test-Connection should complete and continue on with the script.  This cmdlet works correctly when running from command line or non-form based script.  When called from a Windows form, the Test-Connection just hangs the script/form indefinitely.  I have tried various combinations of parameters such as removing "-quiet" and "-count".  Maybe this is not a bug, and I am just overlooking some parameter or alternate method to call this cmdlet.

# Actual behavior
Form hangs; you have to kill the Pwsh.exe process.

# Environment data

```none
Name                           Value
----                           -----
PSVersion                      7.0.0
PSEdition                      Core
GitCommitId                    7.0.0
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by rjmholt** [comment]
Mysteriously, this has been fixed before in https://github.com/PowerShell/PowerShell/issues/11418.

Looks like there's more to do.

**Comment by rjmholt** [comment]
This happens because WinForms constructors set a thread synchronisation context. You might be able to work around it by setting one yourself, but that will likely cause issues with the WinForms app.

@daxian-dbw might have better advice.

**Comment by rjmholt** [comment]
Sorry, actual fix in https://github.com/PowerShell/PowerShell/pull/11517

**Comment by NJK-HOME** [comment]
Ah - sorry, my fault.  I should have done a more thorough search of issues before creating a new issue.  Should I close this issue since it is duplicated?

**Comment by rjmholt** [comment]
No no! It looks like that issue was closed as fixed, but you're hitting it. So this one has value.

**Comment by vexx32** [comment]
@rjmholt I'm sure @daxian-dbw can say for sure, but I don't think this fix was included in the 7.0 GA release. I have a vague recollection that there was talk of including it in a bugfix release hopefully not too long after GA.

**Comment by iSazonov** [comment]
Yes, the fix milestone is 7.1-preview1

**Comment by rjmholt** [comment]
Closing as duplicate
